src/NHibernate/Type/EmbeddedComponentType.cs

```diff
@@ -19,9 +19,12 @@ public override bool IsEmbedded
 
 		public override object Instantiate(object parent, ISessionImplementor session)
 		{
-			bool useParent= false;
-			// NH Different implementation : since we are not sure about why H3.2 use the "parent"
-			//useParent = parent != null && base.ReturnedClass.IsInstanceOfType(parent);
+			var useParent =
+				parent != null &&
+				//TODO: Yuck! This is not quite good enough, it's a quick
```

Comment:
The comment is from here: https://github.com/hibernate/hibernate-orm/blob/5c5de945e56b613b6ed8d0acc060bb124ca23584/hibernate-core/src/main/java/org/hibernate/type/EmbeddedComponentType.java#L34-L37